lib/nimble_options.ex

```diff
@@ -245,14 +290,32 @@ defmodule NimbleOptions do
   This function behaves exactly like `validate/2`, but returns the options directly
   if they're valid or raises a `NimbleOptions.ValidationError` exception otherwise.
   """
-  @spec validate!(keyword(), schema()) :: validated_options :: keyword()
+  @spec validate!(keyword(), schema() | t()) :: validated_options :: keyword()
   def validate!(options, schema) do
     case validate(options, schema) do
       {:ok, options} -> options
       {:error, %ValidationError{} = error} -> raise error
     end
   end
 
+  @doc """
+  Validates the given `schema` and returns a wrapped schema to be used with `validate/2`.
+
+  If the given schema is not valid, raises a `NimbleOptions.ValidationError`.
+  """
+  @spec new!(schema()) :: t() | {:error, ValidationError.t()}
```

Comment:
`{:error, _}` will never be returned here.

```suggestion
  @spec new!(schema()) :: t()
```